src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -69,6 +69,65 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
             return items;
         }
 
+        /// <summary>
+        /// Get the metadata name of all the base types and interfaces this type derived from.
+        /// </summary>
+        private static ImmutableArray<string> GetReceiverTypeNames(ITypeSymbol receiverTypeSymbol)
+        {
+            using var _ = PooledHashSet<string>.GetInstance(out var allTypeNamesBuilder);
+            AddNamesForTypeWorker(receiverTypeSymbol, allTypeNamesBuilder);
+            return allTypeNamesBuilder.ToImmutableArray();
+
+            static void AddNamesForTypeWorker(ITypeSymbol receiverTypeSymbol, PooledHashSet<string> builder)
+            {
+                if (receiverTypeSymbol is ITypeParameterSymbol typeParameter)
+                {
+                    foreach (var constraintType in typeParameter.ConstraintTypes)
+                    {
+                        AddNamesForTypeWorker(constraintType, builder);
+                    }
+                }
+                else
+                {
+                    builder.Add(GetReceiverTypeName(receiverTypeSymbol));
+                    builder.AddRange(receiverTypeSymbol.GetBaseTypes().Select(t => t.MetadataName));
+                    builder.AddRange(receiverTypeSymbol.GetAllInterfacesIncludingThis().Select(t => t.MetadataName));
+
+                    // interface doesn't inherit from object, but is implicitly convertible to object type.
+                    if (receiverTypeSymbol.IsInterfaceType())
+                    {
+                        builder.Add(nameof(Object));
+                    }
+                }
+            }
+        }
+
+        private static string GetReceiverTypeName(ITypeSymbol typeSymbol)
+        {
+            switch (typeSymbol)
+            {
+                case INamedTypeSymbol namedType:
+                    return namedType.IsTupleType ? string.Empty : namedType.MetadataName;
+
+                case IArrayTypeSymbol arrayType:
+                    var elementType = arrayType.ElementType;
+                    while (elementType is IArrayTypeSymbol symbol)
+                    {
+                        elementType = symbol.ElementType;
+                    }
+
+                    var elementTypeName = GetReceiverTypeName(elementType);
+
+                    // We do not differentiate array of different kinds sicne they are all represented in the indices as "NonArrayElementTypeName[]"
+                    // e.g. int[], int[][], int[,], etc. are all represented as "int[]", whereas array of complex type such as T[] is "[]".
+                    return elementTypeName + "[]";
```

Comment:
any concerns about things like `(int, int)[]`?  or `(int[], int[])`?  i.e. mixes of tuples/arrays?

Reply:
if we use the `"[]"` elsewhere, consider a named ocnstant.  or a hel that makes htis name.

Reply:
> any concerns about things like (int, int)[]? or (int[], int[])? i.e. mixes of tuples/arrays?

Those should work now.